Import a Domain flow: Implement new layout

claudiucelfilip opened the pull request:
#### Proposed Changes

This applies the latest layout change for the Import a Domain flow.
<img width="420" alt="Screenshot 2022-11-09 at 13 03 23" src="https://user-images.githubusercontent.com/2749938/200969907-dced4d37-a759-45cc-b770-acec28a45cc6.png">




#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Use an English locale
* Go to `/start/domains/use-your-domain`
* You should see the new design:

<img width="420" src="https://user-images.githubusercontent.com/2749938/200970101-c90b9565-4169-4e00-af9c-9b858752f10a.png"/>


* Use a non-English locale
* Refresh
* You should see the new design but untranslated text should not appear
* Everything else should match the design

<img width="420" alt="Screen Shot on 2022-11-09 at 17-16-13" src="https://user-images.githubusercontent.com/2749938/200970186-40733974-a0e7-4b4e-8516-873ac6398f35.png">


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #69787

claudiucelfilip committed:
Add the latest design changes

delputnam commented:
I'm afraid that pre-filling the "www." will confuse users who want to map a subdomain like "foo.bar.com"

I realize that this will still work, but I think that they might misunderstand and think that they will be mapping "www.foo.bar.com" and NOT (also) "foo.bar.com".

simison commented:
Design looks otherwise great but I have the same feedback as @delputnam — another aspect is that our domains won't have `www.` in them and would be pity if users wouldn't understand that.

Suggesting
- Drop the www from input
- Handle users who enter www. in their domain graciously; currently we just error.
- Either show them gentle notice that the domain will be without www even if they entered it (preferred by me and @delputnam in Slack convo), or
- Handle at backend without a notice to customer

See the original slack convo for better context about the above.

All that can be separate PRs, so without the www — ship it! :-) Wasn't able to review the code at this time.

JanaMW27 commented:
Here are the new designs without the www:

![Import a domain](https://user-images.githubusercontent.com/71137829/200964161-d60fc897-5a73-416a-b89e-83cb3c535790.png)

And if the user puts the domain the wrong way, we could highlight the message in red:
![Import a domain error state](https://user-images.githubusercontent.com/71137829/200964210-47196e78-d37e-420b-817c-c5b81bdbf6d5.png)

claudiucelfilip committed:
Remove fixed www

claudiucelfilip commented:
@simison @delputnam - I removed the fixed `www.` and implemented the note from @JanaMW27's [comment](https://github.com/Automattic/wp-calypso/pull/69898#issuecomment-1309537361)

> Either show them gentle notice that the domain will be without www even if they entered it (preferred by me and @delputnam in Slack convo)

There was the idea of showing the error as the users start typing `www.` - I suggest we follow up on that in a different PR.

delputnam commented:
I think this looks much better now.  I don't quite understand the meaning of "This won't affect your existing site" though.  What are we trying to convey to the user there?

paulbonahora commented:
@delputnam we got feedback that some users are concerned their current site may be affected by inserting their address there. Since it's a domain they already own (and that could be live hosted somewhere else), some are afraid it could interfere with it. That note intends to address that concern. Do you feel that may not be clear and that we'd rather remove that?

delputnam commented:
Oh.  I see what you mean.  It wasn't clear which "site" the message refers to.  I thought it was referring to the WPCOM site.  Maybe we can try to think of a better way to word this.  🤔 

Also, we'll have to make sure that these changes don't affect the other places that this page is used.  See the example below:

<img width="1479" alt="Use_A_Domain_I_Own_‹_A8C_Test_Site_—_WordPress_com" src="https://user-images.githubusercontent.com/1379730/200974282-cdd8a07f-1ef3-436c-bb0e-f61ae4f7b33a.png">

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7619182

Thank you @claudiucelfilip for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7619182).
